core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -141,24 +137,9 @@
             try {
                 final SslContextBuilder builder = SslContextBuilder.forClient();
 
-                builder.sslProvider(
-                        Flags.useOpenSsl() ? SslProvider.OPENSSL : SslProvider.JDK);
+                SslContextUtil.configureDefaults(builder, httpPreference == HttpPreference.HTTP1_REQUIRED);
                 clientFactory.sslContextCustomizer().accept(builder);
 
-                if (httpPreference == HttpPreference.HTTP2_REQUIRED ||
-                    httpPreference == HttpPreference.HTTP2_PREFERRED) {
-
-                    builder.ciphers(Http2SecurityUtil.CIPHERS, SupportedCipherSuiteFilter.INSTANCE)
-                           .applicationProtocolConfig(new ApplicationProtocolConfig(
```

Comment:
Can we make sure these two calls (ciphers and applicationProtocolConfig), or at least applicationProtocolConfig call, are made after the customizer is applied? Otherwise we would end up violating HTTP/2 specification IIUC.

Reply:
Ah thanks - I was leaning towards allowing users to have maximize ability to customize, but makes sense not to let them break HTTP/2 compatibility.